ra/ra.go

```diff
@@ -333,36 +334,70 @@ func (ra *RegistrationAuthorityImpl) checkRegistrationLimits(ctx context.Context
 }
 
 // NewRegistration constructs a new Registration from a request.
-func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, init core.Registration) (core.Registration, error) {
-	if err := ra.keyPolicy.GoodKey(ctx, init.Key.Key); err != nil {
-		return core.Registration{}, berrors.MalformedError("invalid public key: %s", err.Error())
+func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, request *corepb.Registration) (*corepb.Registration, error) {
+	// Func to return false if the key length or ip are zero
+	newRegistrationValid := func(reg *corepb.Registration) bool {
+		return !(len(reg.Key) == 0 || len(reg.InitialIP) == 0)
 	}
-	if err := ra.checkRegistrationLimits(ctx, init.InitialIP); err != nil {
-		return core.Registration{}, err
+	// Error if the request is nil or the registration valid check fails
+	if request == nil || !newRegistrationValid(request) {
+		return nil, errors.New("incomplete gRPC request message")
+	}
+
+	// Convert key bytes to key
+	var key jose.JSONWebKey
+	if err := key.UnmarshalJSON(request.Key); err != nil {
+		return &corepb.Registration{}, berrors.InternalServerError("failed to unmarshal account key: %s", err.Error())
```

Comment:
In all the error cases, we should be returning `nil` instead of `&corepb.Registration{}`.